### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by fredo509**
For this milestone I :  
Create a class `Classroom `.
Create the `has-many/belongs-to` relationship between `Classroom` and ` Student`.
Create a class `Book`.
Create a class ` Rental`
Create the `many-to-many `(also has-many-through) relationship between` Person` and `Book` using the intermediate class `Rental`.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'date'
+
+class Rental
+  attr_accessor :date
+
+  def initialize(book, person, date = DateTime.now())
+    @date = date
+    @book = book
+    @person = person
```
Comment (Danieldotcomcoder) [inline rental.rb:9]: Kindly make sure that the rental is adding itself to the @book to achieve the 'belongs-to' association.
Example : `book.rentals << self`

Also the same for the person

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
+end
```
Comment (Danieldotcomcoder) [inline book.rb:14]: Kindly make sure that rentals has setters and getters method, and inside the add rental method the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.
Example: 
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @fredo509 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @fredo509      👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the use of GitHub features✔️ 
- Kudos on implementing the features suggested  ✔️ 
- Nice use of Ruby syntax 👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**